### PR TITLE
contracts-bedrock: fix gas benchmarks

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Fixes the benchmarks by ensuring that the optimism portal is initialized as well as ensuring that the prev base fee is always explicitly set.

### CODE REVIEW

File: packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol [hunk 1]
```diff
@@ -141,28 +149,46 @@ contract GasBenchMark_L1CrossDomainMessenger is Messenger_Initializer {
...
 
     function test_depositETH_benchmark_1() external {
-        setPrevBaseFee(vm, address(op), INITIAL_BASE_FEE);
+        setPrevBaseFee(vm, address(op), 10 gwei);
         L1Bridge.depositETH{ value: 500 }(50000, hex"");
     }
 
     function test_depositERC20_benchmark_0() external {
-        L1Bridge.depositETH{ value: 500 }(50000, hex"");
+        setPrevBaseFee(vm, address(op), INITIAL_BASE_FEE);
```
Comment (clabby) [inline packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol:173]: Can we use the `pauseGasMetering` and `resumeGasMetering` cheatcodes here to get a more accurate figure?

https://github.com/foundry-rs/forge-std/blob/master/src/Vm.sol#L290-L293

File: packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol [hunk 2]
```diff
@@ -141,28 +149,46 @@ contract GasBenchMark_L1CrossDomainMessenger is Messenger_Initializer {
...
+            _amount: 100,
+            _minGasLimit: 100_000,
+            _extraData: hex""
+        });
     }
 
     function test_depositERC20_benchmark_1() external {
-        setPrevBaseFee(vm, address(op), INITIAL_BASE_FEE);
-        L1Bridge.depositETH{ value: 500 }(50000, hex"");
+        setPrevBaseFee(vm, address(op), 10 gwei);
```
Comment (clabby) [inline packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol:184]: ibid ^

File: packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol [hunk 3]
```diff
@@ -124,15 +124,27 @@ contract GasBenchMark_OptimismPortal is Portal_Initializer {
...
         L1Messenger.sendMessage(bob, data, uint32(100));
     }
 
     function test_sendMessage_benchmark_1() external {
-        setPrevBaseFee(vm, address(op), INITIAL_BASE_FEE);
+        vm.pauseGasMetering();
+        setPrevBaseFee(vm, address(op), 10 gwei);
+        vm.resumeGasMetering();
         // The amount of data typically sent during a bridge deposit.
         bytes
```
Comment (clabby) [inline packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol:149]: ibid ^

File: packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol [hunk 4]
```diff
@@ -124,15 +124,27 @@ contract GasBenchMark_OptimismPortal is Portal_Initializer {
...
+        super.setUp();
+        INITIAL_BASE_FEE = op.INITIAL_BASE_FEE();
+    }
+
     function test_sendMessage_benchmark_0() external {
+        vm.pauseGasMetering();
+        setPrevBaseFee(vm, address(op), INITIAL_BASE_FEE);
+        vm.resumeGasMetering();
         // The amount of data typically sent during a bridge deposit.
         bytes
```
[the remark this answers is not in the document]
Comment (tynes) [reply packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol:139]: good catch altho it does end up in memory as part of the call 
https://github.com/ethereum-optimism/optimism/blob/4071f64e5dd582beaa28c393b20757193361a4ca/packages/contracts-bedrock/contracts/L1/L1CrossDomainMessenger.sol#L49

**Review by clabby — APPROVED** [review]
LGTM

**Pull request merged**